variables.tf

```diff
@@ -252,3 +252,13 @@ variable "bucket_name" {
   default     = null
   description = "Bucket name. If provided, the bucket will be created with this name instead of generating the name from the context"
 }
+
+variable "object_lock_configuration" {
+  type = object({
+    mode  = string # Valid values are GOVERNANCE and COMPLIANCE.
+    days  = number
+    years = number
+  })
+  default     = null
+  description = "A configuration of S3 object locking. With S3 Object Lock, you can store objects using a write-once-read-many (WORM) model. Object Lock can help prevent objects from being deleted or overwritten for a fixed amount of time or indefinitely."
```

Comment:
s/A configuration of/A configuration for

Same comment as above `write once, read many`